model/project_ref.go

```diff
@@ -1372,6 +1395,143 @@ func FindOneProjectRefWithCommitQueueByOwnerRepoAndBranch(owner, repo, branch st
 	return nil, nil
 }
 
+// EnableWebhooks returns true if a hook for the given owner/repo exists or was inserted.
+func EnableWebhooks(ctx context.Context, projectRef *ProjectRef) (bool, error) {
+	hook, err := FindGithubHook(projectRef.Owner, projectRef.Repo)
+	if err != nil {
+		return false, errors.Wrapf(err, "Database error finding github hook for project '%s'", projectRef.Id)
+	}
+	if hook != nil {
+		projectRef.TracksPushEvents = utility.TruePtr()
+		return true, nil
+	}
+
+	settings, err := evergreen.GetConfig()
+	if err != nil {
+		return false, errors.Wrap(err, "error finding evergreen settings")
+	}
+
+	hook, err = SetupNewGithubHook(ctx, *settings, projectRef.Owner, projectRef.Repo)
+	if err != nil {
+		// don't return error:
+		// sometimes people change a project to track a personal
+		// branch we don't have access to
+		grip.Error(message.WrapError(err, message.Fields{
+			"source":             "patch project",
+			"message":            "can't setup webhook",
+			"project":            projectRef.Id,
+			"project_identifier": projectRef.Identifier,
+			"owner":              projectRef.Owner,
+			"repo":               projectRef.Repo,
+		}))
+		projectRef.TracksPushEvents = utility.FalsePtr()
+		return false, nil
+	}
+
+	if err = hook.Insert(); err != nil {
+		return false, errors.Wrapf(err, "error inserting new webhook for project '%s'", projectRef.Id)
+	}
+	projectRef.TracksPushEvents = utility.TruePtr()
+	return true, nil
+}
+
+func EnablePRTesting(projectRef *ProjectRef) error {
```

Comment:
I think this should also be a helper function of the route that calls it, since it's only part of one route (same for EnableCommitQueue), or else this needs to be reworded as "canEnable" since it doesn't actually enable anything